app/publisher/publication_failed.py

```diff
@@ -0,0 +1,2 @@
+class PublicationFailed(Exception):
```

Comment:
This should be in an `exceptions` module: `app/publisher/exceptions.py`

Reply:
Done.